x-pack/filebeat/input/httpjson/request.go

```diff
@@ -347,10 +347,8 @@ func (r *requester) doRequest(stdCtx context.Context, trCtx *transformContext, p
 				return err
 			}
 			// we will only processAndPublishEvents here if chains & root level pagination do not exist, inorder to avoid unnecessary pagination
```

Comment:
This comment is no longer valid. It would be helpful to explain here instead that the interval is only cleared when we don't expect we are building a chain. This is the root of the issue so the comment will be valuable.

Reply:
sure will update that

Reply:
updated the comment